src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -305,12 +314,11 @@ public void maybeTryAdditionalReplicas()
                 speculated = true;
                 cfs.metric.speculativeRetries.inc();
                 // Could be waiting on the data, or on enough digests.
+                Replica extraReplica = Iterables.getLast(targetReplicas);
                 ReadCommand retryCommand = command;
-                if (handler.resolver.isDataPresent())
-                    retryCommand = command.copyAsDigestQuery();
 
-                Replica extraReplica = Iterables.getLast(targetReplicas);
-                Replicas.checkFull(extraReplica);
+                if (handler.resolver.isDataPresent() && extraReplica.isFull())
```

Comment:
Huh, so this kind of just gives up on speculation if we didn't get the data response back? It's always done that as well. Weird.